de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SarosSession.java

```diff
@@ -754,9 +754,20 @@ private void sendActivity(final List<User> recipients,
         }
 
         boolean send = true;
-        // handle FileActivities and FolderActivities to update ProjectMapper
+
+        /*
+         * To explain some magic: When the host generates an activity this
+         * method will be called multiple times. It is first called with the
+         * host as it is only recipient and then it is called again for the
+         * other recipients.
+         */
+        // handle IFileSystemModificationActivities to update ProjectMapper
         if (activity instanceof IFileSystemModificationActivity) {
-            send = updatePartialSharedResources((IFileSystemModificationActivity) activity);
+            if (!isHost() || (isHost() && recipients
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Merge this if statement with the enclosing one. ([squid:S1066](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1066))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=237&issues=AWYXZoXdM63is4RDRNN2&open=AWYXZoXdM63is4RDRNN2)

[ISSUE_KEY]: AWYXZoXdM63is4RDRNN2